### DISCUSSION TITLE
text to image arraybuffer to imageData

### DISCUSSION THREAD

**Discussion by hpssjellis** [huggingface/huggingface.js#93] [Q&A] [answered]
@coyotte508 @radames any suggestions for converting the text to image arrayBuffer to an imageData object. This does not seem to have the correct dimensions

```

      let UAC = new Uint8ClampedArray( objectDetectionRes, 768, 768);
      let imageData = new ImageData(UAC, 768, 768);

```

**Comment by coyotte508** [comment]
Hi @hpssjellis feel free to use discussions for usage questions. @TimMikeladze what do you think about returning a `blob` instead from `@huggingface/inference`, it will then contain the mime type of the image as well?

**Reply by hpssjellis** [reply]
> Hi @hpssjellis feel free to use discussions for usage questions. @TimMikeladze what do you think about returning a `blob` instead from `@huggingface/inference`, it will then contain the mime type of the image as well?

Returning a blob would be useful, but for the moment can anyone tell me the image relevant properties of the arrayBuffer so I can attempt to put the image on a webpage. Presently I assume the uInt8Array is relevant but it can also be int16 or int32

![image](https://user-images.githubusercontent.com/5605614/224750337-5bf5cb26-c45e-479e-be80-fffb8faea68f.png)

**Reply by coyotte508** [reply]
You need to use `img` and convert to a data url: https://stackoverflow.com/questions/8499633/how-to-display-base64-images-in-html

but for that you need to know the mime type (png or jpeg)

**Reply by radames** [reply]
right now @hpssjellis you need to manually set the mime type here is the full examples. But I guess it's a great idea to return the blob as well @coyotte508 !!

```javascript
await hf
  .textToImage({
    inputs:
      "award winning high resolution photo of a giant tortoise/((ladybird)) hybrid, [trending on artstation]",
    negative_prompt: "blurry",
    model: "stabilityai/stable-diffusion-2"
  })
  .then((arrayBuffer) => {
    const image = new Image();
    const blob = new Blob([arrayBuffer], { type: "image/jpeg" });
    image.src = URL.createObjectURL(blob);
    return image;
  })
```

**Reply by coyotte508** [reply]
Alright @hpssjellis you can try the new version `@huggingface/inference@1.6.0`, a blob is directly returned!

You can also draw the `img` to a canvas if you want to get its `ImageData`;

**Reply by radames** [reply]
now you just need to remove the blob line 

```javascript
await hf
  .textToImage({
    inputs:
      "award winning high resolution photo of a giant tortoise/((ladybird)) hybrid, [trending on artstation]",
    negative_prompt: "blurry",
    model: "stabilityai/stable-diffusion-2"
  })
  .then((blob) => {
    const image = new Image();
    image.src = URL.createObjectURL(blob);
    return image;
  })
```

**Reply by hpssjellis** [reply]
> Alright @hpssjellis you can try the new version `@huggingface/inference@1.6.0`, a blob is directly returned!
> 
> You can also draw the `img` to a canvas if you want to get its `ImageData`;

Working on this now. Thanks so much for making the new version. You are fast.

**Comment by hpssjellis** [comment]
Does anyone know if the generated arrayBuffer has an Alpha channel? I think it just has the RGB channels which is why imageData is unhappy. I could just inject an alpha of 255 for every pixel. Any opinions?

**Reply by coyotte508** [reply]
The arrayBuffer doesn't contain raw pixels, it contains a compressed jpg or png. You need to load it in an `img` or use a library like `Jimp` to decode it

**Comment by hpssjellis** [comment]
@coyotte508 @radames Amazing photo quality loads the turtle but strangely doesn't load anything else.


Demo [here](https://hpssjellis.github.io/my-examples-of-huggingfacejs/public/hug16-text-to-image.html)

**Comment by hpssjellis** [comment, marked as answer]
Actually I think it works fine. Just has a bit of a delay that I was too impatient to spot. Thanks for all the help. I only hve one more issue. I will pot it in the discussion.

**Reply by radames** [reply]
yes, it works for me. one reminder the API inference, a cache results based on the same input, and when it's new, it will take many seconds to load the model if it's not on memory yet, then run the inference 

<img width="799" alt="image" src="https://user-images.githubusercontent.com/102277/224793794-96be0ca3-0bbc-4e8c-9dd1-7962773b1a78.png">
